OctoKit/NotificationThread.swift

```diff
@@ -0,0 +1,336 @@
+     - parameter perPage: Number of notifications per page. `100` by default.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func myNotifications(_ session: RequestKitURLSession = URLSession.shared, all: Bool = false, participating: Bool = false,  page: String = "1", perPage: String = "100", completion: @escaping (_ response: Response<[NotificationThread]>) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = NotificationRouter.readNotifications(configuration,all,participating, page, perPage)
+        return router.load(session, dateDecodingStrategy: .formatted(Time.rfc3339DateFormatter), expectedResultType: [NotificationThread].self) { notifications, error in
+            if let error = error {
+                completion(Response.failure(error))
+            } else {
+                if let notifications = notifications {
+                    completion(Response.success(notifications))
+                }
+            }
+        }
+    }
+    
+    /**
+     Marks All Notifications As read
+     - parameter session: RequestKitURLSession, defaults to URLSession.sharedSession()
+     - parameter lastReadAt: Describes the last point that notifications were checked `last_read_at` by default.
+     - parameter read: Whether the notification has been read `false` by default.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func markNotificationsRead(_ session: RequestKitURLSession = URLSession.shared,lastReadAt: String = "last_read_at",read: Bool = false, completion: @escaping (_ response: Error?) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = NotificationRouter.markNotificationsRead(configuration, lastReadAt, read)
+        return router.load(session,completion: completion)
+    }
+    
+    /**
+     Marks All Notifications As read
+     - parameter session: RequestKitURLSession, defaults to URLSession.sharedSession()
+     - parameter threadId: The ID of the Thread.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func getNotificationThread(_ session: RequestKitURLSession = URLSession.shared, threadId: String, completion: @escaping (_ response: Response<NotificationThread>) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = NotificationRouter.getNotificationThread(configuration,threadId)
+        return router.load(session, dateDecodingStrategy: .formatted(Time.rfc3339DateFormatter), expectedResultType: NotificationThread.self) { notification, error in
+            if let error = error {
+                completion(Response.failure(error))
+            } else {
+                if let notification = notification {
+                    completion(Response.success(notification))
+                }
+            }
+        }
+    }
+    
+    //TODO: mark thread as read function, requires the PATCH update to RequestsKit
```

Comment:
You can remove the `TODO` or resolve it. Please also remove all the empty lines except for one.